gnpy/core/equipment.py

```diff
@@ -0,0 +1,119 @@
+    This is to check that the nf_min and nf_max values from the json file
+    make sense. If not a warning is printed """
+    if nf1 < 4:
+        print('1st coil nf calculated value {} is too low: revise inputs'.format(nf1))
+    if nf2 < nf1 + 0.3 or nf2 > nf1 + 2: 
+        """nf2 should be with [nf1+0.5 - nf1 +2] boundaries
+        there shouldn't be very high nf differences between 2 coils
+        => recalculate delta_p 
+        """            
+        nf2 = max(nf2, nf1+0.3)
+        nf2 = min(nf2, nf1+2)
+        g1a_max = lin2db(db2lin(nf2) / (db2lin(nf_min) - db2lin(nf1))) #use expression (1)
+        delta_p = gain_max - g1a_max
+        g1a_min = gain_min - (gain_max-gain_min) - delta_p
+        if delta_p < 1 or delta_p > 6:
+            #delta_p should be > 1dB and < 6dB => consider user warning if not
+            print('1st coil vs 2nd coil calculated DeltaP {} is not valid: revise inputs'
+                        .format(delta_p))
+    #check the calculated values for nf1 & nf2:
+    nf_min_calc = lin2db(db2lin(nf1) + db2lin(nf2)/db2lin(g1a_max))
+    nf_max_calc = lin2db(db2lin(nf1) + db2lin(nf2)/db2lin(g1a_min))
+    if (abs(nf_min_calc-nf_min) > 0.01) or (abs(nf_max_calc-nf_max) > 0.01):
+        print('nf model calculation failed with nf_min {} and nf_max {} calculated'
+                .format(nf_min_calc, nf_max_calc))
+
+    return (nf1, nf2, delta_p)
+
+
+def read_eqpt_library(filename):
+    """build global dictionnary eqpt_library that stores all eqpt characteristics:
+    edfa type type_variety, fiber type_variety
+    from the eqpt_config.json (filename parameter)
+    also read advanced_config_from_json file parameters for edfa if they are available:
+    typically nf_ripple, dfg gain ripple, dgt and nf polynomial nf_fit_coeff
+    if advanced_config_from_json file parameter is not present: use nf_model:
+    requires nf_min and nf_max values boundaries of the edfa gain range
+    """
+    global eqpt_library
+    eqpt_library = load_json(filename)
+    for i, el in enumerate(eqpt_library['Edfa']): 
+        dict_nf_model = {}
+        if 'advanced_config_from_json' in el:
+            #use advanced amplifier model with full ripple characterization
+            config_json_filename = el.pop(ADVANCED_CONFIG_JSON_FILENAME)
+            dict_nf_model['nf_model'] = {'enabled': False}
+        else:
+            #use a default ripple model (only default dgt is defined)
+            config_json_filename = DEFAULT_CONFIG_JSON_FILENAME
+            (nf1, nf2, delta_p) = nf_model(el)
+            #remove nf_min and nf_max field and replace by nf1, nf2 & delta_p
+            nf_min = el.pop('nf_min','')
+            nf_max = el.pop('nf_max','')
+            dict_nf_model['nf_model'] = dict(zip(["enabled","nf1","nf2","delta_p"],[True,nf1,nf2,delta_p]))
+
+        json_data = load_json(config_json_filename)
+        eqpt_library['Edfa'][i] = {**el, **json_data, **dict_nf_model}
```

Comment:
It's probably better to return a new structure computed from the old structure rather than trying to change the old structure in-line.

Instead of:
```
obj = ...
for i, el in enumerate(obj):
  obj[i] = ...
```

You can do:
```
raw = ...
clean = ...
for el in raw:
   clean.append( ... ) # or clean[...] = ...
return clean
```